### DISCUSSION TITLE
Release-7.0.0 issues while deploying using release notes

### DISCUSSION THREAD

**Discussion by sanketnagdive** [Sunbird-Knowlg/Community#177] [General Discussions]
This discussion thread created to highlight the challenges while deploying to the staging environment. cc. @AmiableAnil @vinukumar-vs  @ashokreddy1208 @gandham-santhosh @amitpriyadarshi @rajeevsathish @mohitga @pallakartheekreddy

**Comment by sanketnagdive** [comment]
@pallakartheekreddy  As per instructions we are using release note 5.6.0 and 5.7.0 respt issues we faced are
1. The repo and branch given for sunbird-learning-sevice is not working for deployment. 
2. for taxonomy build CSP support variables are not listed to be added to jenkins build job, I have checked in dev env and added the same for deployment. 
Please confirm the same. cc @AmiableAnil @vinukumar-vs @ashokreddy1208 @gandham-santhosh @amitpriyadarshi @rajeevsathish @mohitga

**Comment by mohitga** [comment]
@pallakartheekreddy
can you unblock @sanketnagdive here?

**Reply by pallakartheekreddy** [reply]
@sanketnagdive This issue fixed right? Can you please update how this issue fixed...
@mohitga

**Reply by sanketnagdive** [reply]
@pallakartheekreddy Thanks for the help. It is fixed and deployed on staging. CSP varibles added to the jenkins job for build taxonomy and taxonomy has been deployed with branch release-7.0.0, for 2nd issue compared the changes of repo and tag from the listed release note and tried deploying with branch release-5.6.0. Replicating same for cokreat  cc. @mohitga

**Reply by mohitga** [reply]
Thanks @sanketnagdive,

@pallakartheekreddy can you take the changes done by @sanketnagdive  and update the release note and relevant documentation.

**Reply by pallakartheekreddy** [reply]
@mohitga Release notes, already updated with details.

**Reply by pallakartheekreddy** [reply]
@sanketnagdive Please close this thread as the issue already resolved.

**Reply by sanketnagdive** [reply]
@pallakartheekreddy we are still facing issue with taxonomy service.

**Reply by pallakartheekreddy** [reply]
@sanketnagdive Can you share more details?

**Reply by sanketnagdive** [reply]
![Screenshot from 2024-01-19 10-47-54](https://github.com/Sunbird-Knowlg/Community/assets/31030038/dd5f58ed-cb32-481e-b8f3-b780f47c12a6)
@pallakartheekreddy

**Reply by pallakartheekreddy** [reply]
@sanketnagdive did you add below configuration in taxonomy service application conf?
https://knowlg.sunbird.org/use/release-notes/release-5.6.0#knowledge-platform-greater-than-taxonomy-service

**Reply by sanketnagdive** [reply]
This are added. The problem is when we deploy we able to access framework api but after some time its stops and need to restart. Is there any specific xmx size we need to use. We tried increaing memory for the service but its not working.

**Reply by aimansharief** [reply]
@sanketnagdive can you please test by adding these as environment variables
```
JAVA_OPTIONS: -Xms500m -Xmx500m
_JAVA_OPTIONS: -Dlog4j2.formatMsgNoLookups=true
```

**Reply by sanketnagdive** [reply]
@pallakartheekreddy @aimansharief we tried all the above options. framework read api is working but now framework publish is failing. We have tried allocating more resoures to taxonomy services, api call is restarting the service. Earlier neo4j was falling as the process is consuming high memory usage so we tried increasing the memory for the VM even after its consuming high memory usage. We can setup a call.

**Reply by sanketnagdive** [reply]
attaching memory and cpu graphs for taxonomy. 
![Screenshot from 2024-01-22 15-25-57](https://github.com/Sunbird-Knowlg/Community/assets/31030038/124fb44f-1a82-4d66-abb4-9d540b4b4069)

**Reply by sanketnagdive** [reply]
@pallakartheekreddy @aimansharief Issue was not related to the taxonomy. it was failing for one specific publish framework and was making neo4j and taxonomy to crash. Amit is looking further for that specific framework. Thanks for the support. cc  @amitpriyadarshi
